src/borg/helpers/datastruct.py

```diff
@@ -49,3 +49,58 @@ def get(self, size=None, init=False):
         if size is not None:
             self.resize(size, init)
         return self.buffer
+
+
+class EfficientBytesQueue:
+    """
+    An efficient FIFO queue that splits received elements into chunks.
+    """
+
+    class SizeUnderflow(Error):
+        """Could not pop_front first {} elements, collection only has {} elements.."""
+
+    def __init__(self, split_size):
+        self.buffers = []
+        self.size = 0
+        self.split_size = split_size
+
+    def peek_front(self):
+        if not self.buffers:
+            return []
+        buffer = self.buffers[0]
```

Comment:
this is a bit inconsistent: returns an empty `list` for empty case, otherwise returns an `element (bytes)`.